runtime/domain.c

```diff
@@ -930,7 +940,7 @@ CAMLprim value caml_domain_spawn(value callback, value mutex)
 CAMLprim value caml_ml_domain_id(value unit)
 {
   CAMLnoalloc;
-  return Val_int(domain_self->interruptor.unique_id);
+  return Val_long(domain_self->interruptor.unique_id);
 }
 
 CAMLprim value caml_ml_domain_unique_token (value unit)
```

Comment:
As I was looking at this, I noticed that `caml_ml_domain_unique_token` is a relic that can be removed:
 https://github.com/ocaml/ocaml/commit/f0e76441549719f126d4d3c7e0aafb6cd2686764

Reply:
Note: I tried to get rid of it naively, but the build breaks because a primitive was removed. I think my bootstrap-fu is not strong enough yet to do this easily (there is some documentation for how to do this in BOOTSTRAP.adoc, but I was hoping that it wasn't necessary to bootstrap anymore thanks to the `-use-prims` flag, which I couldn't find a noob-friendly explanation for.) From the fact that it simply returns `Val_unit`, I suspect that the person who worked on it last hit the same barrier and decided to just remove the implementation and keep the primitive. Some other time!